src/PagePlayers.res

```diff
@@ -337,6 +337,303 @@ module NewPlayerForm = {
+    } else {
+      let errors = nonEmptyResults->Array.keepMap(r =>
+        switch r {
+        | Error(e) => Some(e)
+        | Ok(_) => None
+        }
+      )
+
+      if Array.length(errors) > 0 {
+        Invalid(errors)
+      } else {
+        let players = nonEmptyResults->Array.keepMap(r =>
+          switch r {
+          | Ok(p) => Some(p)
+          | Error(_) => None
+          }
+        )
+        Valid(players)
+      }
+    }
+  }
+
+  @react.component
+  let make = (~dispatch, ~onSuccess=?) => {
+    let (input, setInput) = React.useState(() => "")
+    let (parseResult, setParseResult) = React.useState(() => None)
+    let (importing, setImporting) = React.useState(() => false)
+
+    let handleFileUpload = event => {
+      module FileReader = Externals.FileReader
+      ReactEvent.Form.preventDefault(event)
+      let reader = FileReader.make()
+
+      let onload = ev => {
+        let content = ev["target"]["result"]
+        setInput(_ => content)
+        setParseResult(_ => None)
+      }
+
+      FileReader.setOnLoad(reader, onload)
+      FileReader.readAsText(
+        reader,
+        ReactEvent.Form.currentTarget(event)["files"]->Array.get(0)->Option.getWithDefault(""),
+      )
+      // Clear the filename so the same file can be uploaded again
+      ReactEvent.Form.currentTarget(event)->Object.set("value", "")
+    }
+
+    let handleSubmit = event => {
+      ReactEvent.Form.preventDefault(event)
+
+      switch parseInput(input) {
+      | Invalid(errors) => setParseResult(_ => Some(Invalid(errors)))
+      | Valid(players) => {
+          setImporting(_ => true)
+
+          // Import all players
+          players->Array.forEach(p => {
+            let id = Data.Id.random()
+            dispatch(Db.Set(id, {
+              Player.firstName: p.firstName,
+              lastName: p.lastName,
+              rating: p.rating,
+              id,
+              type_: Person,
+              matchCount: Player.NatInt.fromInt(0),
+            }))
+          })
+
+          setParseResult(_ => Some(Valid(players)))
+          setImporting(_ => false)
+
+          // Call success callback if provided
+          switch onSuccess {
+          | None => ()
+          | Some(fn) => fn()
+          }
+        }
+      }
+    }
+
+    let handlePreview = event => {
+      ReactEvent.Mouse.preventDefault(event)
+      setParseResult(_ => Some(parseInput(input)))
+    }
+
+    <form onSubmit=handleSubmit>
+      <fieldset>
+        <legend> {React.string("Bulk import players")} </legend>
+        <p>
+          <label htmlFor="bulk-import-file">
+            {React.string("Upload a file (CSV, TSV, or TXT):")}
+          </label>
+        </p>
+        <p>
+          <input
+            type_="file"
+            id="bulk-import-file"
+            accept=".csv,.tsv,.txt"
+            onChange=handleFileUpload
+          />
+        </p>
+        <p className="caption-20" style={ReactDOM.Style.make(~textAlign="center", ())}>
```

Comment:
I believe the ` ReactDOM.style` API should be replaced with a simple object with the latest version of Rescript (I'm no expert though!)
```suggestion
        <p className="caption-20" style={{textAlign: "center"}}>
```